taskcluster/sync/Auth.py

```diff
@@ -0,0 +1,511 @@
+    a role.
+
+    As in scopes the `*` kleene star also have special meaning if it is
+    located at the end of a `roleId`. If you have a role with the following
+    `roleId`: `my-prefix*`, then any client which has a scope staring with
+    `assume:my-prefix` will be allowed to assume the role.
+
+    As previously mentioned each client gets the scope:
+    `assume:client-id:<clientId>`, it trivially follows that you can create a
+    role with the `roleId`: `client-id:<clientId>` to assign additional
+    scopes to a client. You can also create a role `client-id:user-*`
+    if you wish to assign a set of scopes to all clients whose `clientId`
+    starts with `user-`.
+
+    ### Guarded Services
+    The authentication service also has API end-points for delegating access
+    to some guarded service such as AWS S3, or Azure Table Storage.
+    Generally, we add API end-points to this server when we wish to use
+    TaskCluster credentials to grant access to a third-party service used
+    by many TaskCluster components.
+    '''
+    version = 0
+    referenceUrl = 'http://references.taskcluster.net/auth/v1/api.json'
+    urls = {
+        'listClients': '{baseUrl}/clients/',
+        'client': '{baseUrl}/clients/{clientId}',
+        'createClient': '{baseUrl}/clients/{clientId}',
+        'resetAccessToken': '{baseUrl}/clients/{clientId}/reset',
+        'updateClient': '{baseUrl}/clients/{clientId}',
+        'enableClient': '{baseUrl}/clients/{clientId}/enable',
+        'disableClient': '{baseUrl}/clients/{clientId}/disable',
+        'deleteClient': '{baseUrl}/clients/{clientId}',
+        'listRoles': '{baseUrl}/roles/',
+        'role': '{baseUrl}/roles/{roleId}',
+        'createRole': '{baseUrl}/roles/{roleId}',
+        'updateRole': '{baseUrl}/roles/{roleId}',
+        'deleteRole': '{baseUrl}/roles/{roleId}',
+        'expandScopes': '{baseUrl}/scopes/expand',
+        'currentScopes': '{baseUrl}/scopes/current',
+        'awsS3Credentials': '{baseUrl}/aws/s3/{level}/{bucket}/{prefix}',
+        'azureTableSAS': '{baseUrl}/azure/{account}/table/{table}/read-write',
+        'authenticateHawk': '{baseUrl}/authenticate-hawk',
+        'testAuthenticate': '{baseUrl}/test-authenticate',
+        'testAuthenticateGet': '{baseUrl}/test-authenticate-get/',
+        'ping': '{baseUrl}/ping',
+    }
+
+    def __init__(self, *args, **kwargs):
+        self.classOptions = {}
+        self.classOptions['baseUrl'] = 'https://auth.taskcluster.net/v1'
+        super(Auth, self).__init__(*args, **kwargs)
+
+    def listClients(self, signUrl=False):
+        '''
+        List Clients
+
+        Get a list of all clients.  With `prefix`, only clients for which
+        it is a prefix of the clientId are returned.
+
+        This method takes no arguments.
+        '''
+        url = self.urls['listClients'].format(
+            baseUrl=self.options['baseUrl'],
+        )
+        if signUrl:
+            url = self.buildSignedUrl(url)
```

Comment:
IIRC, I asked in #taskcluster which methods would need to support signed urls, and you told me it's optional for gets, and the others don't support it.  Because of that, I added the capability to sign the url for all gets.

I don't have any other criteria for which make sense to sign or not.  Are there ways for me to programmatically decide which should support signed urls?